modules/fedimint-ln-client/src/lib.rs

```diff
@@ -975,29 +966,37 @@ impl LightningClientModule {
         if !gateways.is_empty() {
             debug!("Fetching meta overrides from remote source/cache");
             let config = self.client_ctx.get_config().clone();
-            let federation_meta = match self.fetch_meta_overrides(config).await {
-                Ok(meta) => {
-                    if meta.is_empty() {
-                        debug!("No meta overrides found");
-                        return Ok(gateways);
+            let global_meta = config.global.meta.clone();
+            let federation_id = config.global.federation_id();
+
+            let meta = match FederationInfo::from_config(config)
+                .await?
+                .meta::<String>(META_OVERRIDE_URL_KEY)?
+            {
+                Some(override_src) => {
+                    match self.fetch_meta_overrides(federation_id, override_src).await {
+                        Ok(override_meta) => override_meta,
+                        Err(e) => {
+                            error!("Error fetching meta overrides: {}", e);
+                            global_meta
+                        }
                     }
-                    meta
                 }
-                Err(e) => {
-                    error!("Error fetching meta overrides: {}", e);
-                    return Ok(gateways);
+                None => {
+                    debug!("No meta override source configured");
+                    global_meta
                 }
             };
 
+            if meta.is_empty() {
+                debug!("No meta overrides found");
+                return Ok(gateways);
+            }
```

Comment:
This return path seems redundant?

Reply:
yes. need to refactor this a bit more, and there's a bug with the json parsing to fix